utils/pre-publish-check-ci.js

```diff
@@ -19,6 +20,19 @@ fg(path.join(__dirname, "..", "packages", "**", "package.json")).then(
         for (const pkgPath of pkgPaths) {
             const pkgJson = require(path.relative(__dirname, pkgPath));
 
+            // Skip over packages don't have a build step
+            if (pkgJson.nobuild) {
+                const mainPath = path.join(path.dirname(pkgPath), pkgJson.main);
+                if (!fs.existsSync(mainPath)) {
+                    const {name, main} = pkgJson;
+                    console.error(
+                        `ERROR: ${name}'s "main" field is set to ${main}, but no file exists at that path.`,
+                    );
+                    process.exit(1);
+                }
+                continue;
+            }
+
```

Comment:
Could you make this a function like the other checks? Just for readability. Something like `checkNoBuildMain`?

Reply:
Also, should this be in the `!checkPrivate` condition?